packages/controller/src/lib/zipFiles.ts

```diff
@@ -1,55 +1,70 @@
-'use strict';
+import { tools } from '@iobroker/js-controller-common';
+import JSZip from 'jszip';
+import type { Client as ObjectsClient } from '@iobroker/db-objects-redis';
 
-let JSZip;
-const { tools } = require('@iobroker/js-controller-common');
-
-function _getAllFilesInDir(objects, id, name, options, callback, result) {
-    result = result || [];
+function _getAllFilesInDir(
+    objects: ObjectsClient,
+    id: string,
+    name: string,
+    options: any,
+    callback: (errs: null | string[], res: string[]) => void,
+    result?: string[]
+): void {
     objects.readDir(id, name, options, (err, files) => {
+        result = result || [];
+
         let count = 0;
-        const errors = [];
+        const errors: string[] = [];
         if (files) {
-            for (let f = 0; f < files.length; f++) {
-                if (files[f].isDir) {
+            for (const file of files) {
+                if (file.isDir) {
                     count++;
                     _getAllFilesInDir(
                         objects,
                         id,
-                        `${name}/${files[f].file}`,
+                        `${name}/${file.file}`,
                         options,
-                        (err, _result) => {
-                            err && errors.push(err);
+                        (errs, _result) => {
+                            errs && errors.push(...errs);
```

Comment:
but it is always only one result here? https://github.com/ioBroker/ioBroker.js-controller/pull/2124/files#diff-f9e06effa6b21aa59a14649919b29124f62dd68fc6271a3422680960521ea421R131 ... or do I miss anything?

Reply:
it is called recursively I really don't like how all this is done, but didn't want to waste too much time.

Reply:
So you are calling the method, then it initializes an array, it iterates e.g. 10 times, calling the method recursively, collection all the errors in an array.

Reply:
And as the method itself could also do some recursive shit again, you cannot be sure that it is returning one entry, two etc.

Reply:
Ok